File: app/src/main/java/com/fonteraro/martaco/cinemahub/data/remote/ApiInterface.java
package com.fonteraro.martaco.cinemahub.data.remote;

import androidx.annotation.NonNull;

import com.fonteraro.martaco.cinemahub.BuildConfig;
import com.fonteraro.martaco.cinemahub.data.model.db.MovieDetails;
import com.fonteraro.martaco.cinemahub.data.remote.response.MoviesData;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiInterface {

    @GET(BuildConfig.MOVIES)
    Call<MoviesData> getMovies();

    @GET(BuildConfig.MOVIES + "/{ID}")
    Call<MovieDetails> getMovie(@Path("ID") @NonNull String id);

}

File: app/src/main/java/com/fonteraro/martaco/cinemahub/data/remote/ApiInterface.kt
package com.fonteraro.martaco.cinemahub.data.remote

import androidx.annotation.NonNull

import com.fonteraro.martaco.cinemahub.BuildConfig
import com.fonteraro.martaco.cinemahub.data.remote.response.ApiMovie
import com.fonteraro.martaco.cinemahub.data.remote.response.ApiMovies

import retrofit2.http.GET
import retrofit2.http.Path

interface ApiInterface {

    @GET(BuildConfig.MOVIES)
    suspend fun getMovies(): ApiMovies

    @GET("${BuildConfig.MOVIES}/{ID}")
    suspend fun getMovie(@Path("ID") @NonNull id: String): ApiMovie

}
